### PR TITLE
Small clean up in Particle and DomParticle

### PR DESCRIPTION
**Pull request by mykmartin**
Remove unused SlotChanges and StateChanges from Particle.
Move ViewChanges logic into DomParticle.

### CODE REVIEW

**Comment by shans** [comment]
ViewChanges and the other *Changes classes were intended to help people that wanted to build directly on Particle.js. - I didn't reallze DomParticle was using ViewChanges to provide props. Maybe we could put them into a utility file?

File: runtime/dom-particle.js [hunk 1]
```diff
@@ -99,9 +95,14 @@ class DomParticle extends XenStateMixin(Particle) {
     this.handles = views;
     this._views = views;
     let config = this.config;
-    this.when([new ViewChanges(views, config.views, 'change')], async () => {
-      await this._handlesToProps(views, config);
-    });
+
```
Comment (sjmiles) [inline runtime/dom-particle.js]: Unrolling the changes code into this module seems like a step backward, what's the value add?

**Comment by sjmiles** [comment]
I guess my view was that DomParticle itself was being _helped by *Changes to build on Particle_ (which seemed to match the intent).

Are the *Changes being deprecated? If so why? If not, then why unroll that code directly into DomParticle (aka SimpleParticle)?

When you say "put them into a utility file" do you mean put *Changes into their own module? Sure, that would be fine (and it's probably an improvement if it's a more obvious module name for *Changes classes).

File: runtime/dom-particle.js (hunk 1 above)
Comment (mykmartin) [reply runtime/dom-particle.js]: If ViewChanges are going to be of use elsewhere/in the future, then I agree that we shouldn't do this. If it ends up being an API that only DomParticle needs, then the logic should live in DomParticle.

Ultimately I was trying to simplify the Particle code - pruning stuff. Apologies if it's too early to judge that.

**Comment by shans** [comment]
The *Changes were an early attempt to provide a convenience API on top of the different kinds of changes that can happen. Slot- and StateChanges aren't used by DomParticle because DomParticle does its own rendering and state management.

Removing these probably doesn't matter that much; I've got a mild preference for keeping them, or alternatively for fast-following the update-driven API with something that has equivalent functionality. This API should definitely layer on top of the basic particle rather than being part of it.

The basic aim is to get to a state where writing particles without having to buy into the react-like view of the world is at least not incredibly painful :)

**Comment by mykmartin** [comment]
Ok, I'll drop the PR.

**Comment by sjmiles** [comment]
@mykmartin I wholly support any efforts to simplify, I didn't mean to rain on this particular parade, it just seemed like there were implications for fate of *Changes that I wanted to understand. Moving that API to it's own module might be a good first step. 

My intent was always for SimpleParticle (nee DomParticle) to just be one possible flavor (it's the kind I like). Maybe there is a sense of lock-in because Particle doesn't have the template stuff? We could separate the template/slot rendering from the props/state stuff if that would feel more flexible.

**Pull request closed without merging**